packages/paypal-commerce-integration/src/PaypalCommerceRatePayPaymentMethod.tsx

```diff
@@ -83,8 +85,20 @@ const PaypalCommerceRatePayPaymentMethod: FunctionComponent<any> = ({
                     legalTextContainer: 'legal-text-container',
                     getFieldsValues: () => fieldsValues.current,
                     onPaymentSubmission: (isSubmitting: boolean) => setIsPaymentSubmitting(isSubmitting),
-                    onError: (error: Error) => {
+                    onError: (error: SpecificError) => {
                         paymentForm.disableSubmit(method, true);
+                        const ratepaySpecificError = error?.errors?.filter(e => e.provider_error);
+
+                        if (ratepaySpecificError) {
+                           const ratepayError = new CustomError({
+                                data: {
+                                    shouldBeTranslatedAsHtml: true,
+                                    translationKey: `payment.ratepay.errors.${ratepaySpecificError[0]?.provider_error?.code}`
+                                },
+                            });
+                            return onUnhandledError(ratepayError);
+                        }
```

Comment:
It might be better to detect specific error codes that we know about right now. PayPal team can change the error code on their side, and we will get a translation key error in this case.